docs: Fix issues found in Queeny's test (#507)
pingcap/tidb-operator: deploy/aws/README.md

  The `awscli` must be [configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html) before it can interact with AWS. The fastest way to set up is using the `aws configure` command:

  ``` shell
  # Replace AWS Access Key ID and AWS Secret Access Key with your own keys
  $ aws configure
  AWS Access Key ID [None]: AKIAIOSFODNN7EXAMPLE
  AWS Secret Access Key [None]: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  Default region name [None]: us-west-2
  Default output format [None]: json
  ```
  > **Note:** The access key must have at least permissions to: create VPC, create EBS, create EC2 and create role
* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl) >= 1.11
* [helm](https://github.com/helm/helm/blob/master/docs/install.md#installing-the-helm-client) >= 2.9.0
* [jq](https://stedolan.github.io/jq/download/)
* [aws-iam-authenticator](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html) installed in `PATH`, to authenticate with AWS

  The easist way to install `aws-iam-authenticator` is to download the prebuilt binary:

  ``` shell

>   curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.12.7/2019-03-27/bin/linux/amd64/aws-iam-authenticator

Reviewer: I think the AWS deployment can be done on both Linux and macOS. But this command only works for Linux.

# Deploy TiDB Operator and TiDB cluster on AWS EKS

> 

Reviewer 2: ```suggestion

This document describes how to deploy TiDB Operator and a TiDB cluster on AWS EKS with your laptop (Linux or macOS) for development or testing.

```

> ## Requirements

Reviewer 2: Reference doc: [Deploy TiDB to Kubernetes on Your Laptop](https://github.com/pingcap/tidb-operator/blob/d04d7088731a5f4d21f833c1bea8dcc159d64f88/docs/local-dind-tutorial.md)

* [helm](https://github.com/helm/helm/blob/master/docs/install.md#installing-the-helm-client) >= 2.9.0
* [jq](https://stedolan.github.io/jq/download/)
* [aws-iam-authenticator](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html) installed in `PATH`, to authenticate with AWS

  The easiest way to install `aws-iam-authenticator` is to download the prebuilt binary:

  ``` shell
  # Download binary for Linux
  curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.12.7/2019-03-27/bin/linux/amd64/aws-iam-authenticator

  # Or, download binary for macOS
  curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.12.7/2019-03-27/bin/darwin/amd64/aws-iam-authenticator

  chmod +x ./aws-iam-authenticator
  sudo mv ./aws-iam-authenticator /usr/local/bin/aws-iam-authenticator
  ```

## Deploy

The default setup will create a new VPC and a t2.micro instance as bastion machine, and an EKS cluster with the following ec2 instances as worker nodes:

* 3 m5d.xlarge instances for PD
* 3 i3.2xlarge instances for TiKV
* 2 c4.4xlarge instances for TiDB
* 1 c5.xlarge instance for monitor

> ``` shell

Reviewer 2: Please add an introductory sentence before the shell commands to clarify what these commands do.

* 3 m5d.xlarge instances for PD
* 3 i3.2xlarge instances for TiKV
* 2 c4.4xlarge instances for TiDB
* 1 c5.xlarge instance for monitor

Use the following commands to set up the cluster:

``` shell
# Get the code
$ git clone --depth=1 https://github.com/pingcap/tidb-operator
$ cd tidb-operator/deploy/aws

# Apply the configs, note that you must answer "yes" to `terraform apply` to continue
$ terraform init
$ terraform apply
```

It might take 10 minutes or more to finish the process. After `terraform apply` is executed successfully, some useful information is printed to the console.

A successful deployment will give the output like:

```
Apply complete! Resources: 67 added, 0 changed, 0 destroyed.

Outputs:

bastion_ip = [
    52.14.50.145
]
eks_endpoint = https://E10A1D0368FFD6E1E32E11573E5CE619.sk1.us-east-2.eks.amazonaws.com
eks_version = 1.12
monitor_endpoint = http://abd299cc47af411e98aae02938da0762-1989524000.us-east-2.elb.amazonaws.com:3000
region = us-east-2
tidb_dns = abd2e3f7c7af411e98aae02938da0762-17499b76b312be02.elb.us-east-2.amazonaws.com
tidb_port = 4000
tidb_version = v3.0.0-rc.1
```

> **Note:** You can use the `terraform output` command to get the output again.

> ## Access the DB

Reviewer 2: ```suggestion
## Access the database
```

  The easist way to install `aws-iam-authenticator` is to download the prebuilt binary:

  ``` shell
  # Download binary for Linux
  curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.12.7/2019-03-27/bin/linux/amd64/aws-iam-authenticator

  # Or, download binary for macOS
  curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.12.7/2019-03-27/bin/darwin/amd64/aws-iam-authenticator

> 

Reviewer 3: Maybe using the code below is simpler and more general
```
os=$(uname -s| tr '[:upper:]' '[:lower:]')
curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.12.7/2019-03-27/bin/$os/amd64/aws-iam-authenticator
```

Author: I don't have any mac for test, if you can confirm this works for macOS?

Reviewer 3: Yes, it works for macOS and Linux

Reviewer: @AstroProfundis The [link](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html) you mentioned already has an elegant way to install aws-iam-authenticator.
For macOS, install with homebrew. For linux install with the curl to fetch the binary. It's inappropriate to provide the tricky script in the guide.

Author: I don't think homebrew is available to every macOS installation by default...

Reviewer: Yes, it's not builtin on macOS. But almost all technical users install that. So I think provide homebrew command is reasonable.

Author: This definitely can't pass Queeney's test, the "almost all technical users" is not fitting the target user groups.

Reviewer 2: You can add a note in the document that before users install aws-iam-authenticator with homebrew, they need to make sure that they have installed homebrew first.

Author: Then we're introducing new dependency and copying almost all major parts of AWS document, what's the point of avoiding out-linking again?

Reviewer 2: Yeah, you are right. So if the curl way does not require any new dependency for macOS, I suggest we introduce the curl way in our document then. If users prefer the homebrew way, they can still click the link and find the guide in the AWS document.

* 2 c4.4xlarge instances for TiDB
* 1 c5.xlarge instance for monitor

``` shell
# Get the code
$ git clone --depth=1 https://github.com/pingcap/tidb-operator
$ cd tidb-operator/deploy/aws

# Apply the configs, note that you must answer "yes" to `terraform apply` to continue
$ terraform init
$ terraform apply
```

It might take 10 minutes or more for the process to finish. After `terraform apply` is executed successfully, some useful information is printed to the console. You can access the `monitor_endpoint` address (printed in output) using your web browser to view monitoring metrics.

A successful deploy will print output like:

```
Apply complete! Resources: 67 added, 0 changed, 0 destroyed.

Outputs:

bastion_ip = [
    52.14.50.145
]
eks_endpoint = https://E10A1D0368FFD6E1E32E11573E5CE619.sk1.us-east-2.eks.amazonaws.com
eks_version = 1.12
monitor_endpoint = http://abd299cc47af411e98aae02938da0762-1989524000.us-east-2.elb.amazonaws.com:3000
region = us-east-2
tidb_dns = abd2e3f7c7af411e98aae02938da0762-17499b76b312be02.elb.us-east-2.amazonaws.com
tidb_port = 4000
tidb_version = v3.0.0-rc.1
```

> **Note:** You can use the `terraform output` command to get the output again.

> To access TiDB cluster, use the following command to first ssh into the bastion machine, and then connect it via MySQL client (replace the `<>` parts with values from the output):

Reviewer 2: ```suggestion
To access the deployed TiDB cluster, use the following commands to first `ssh` into the bastion machine, and then connect it via MySQL client (replace the `<>` parts with values from the output):
```

``` shell
ssh -i credentials/k8s-prod-<cluster_name>.pem ec2-user@<bastion_ip>
mysql -h <tidb_dns> -P <tidb_port> -u root
```

The default value of `cluster_name` is `my-cluster`. If the DNS name is not resolvable, be patient and wait a few minutes.

> You can interact with the EKS cluster using `kubectl` and `helm` with the kubeconfig file `credentials/kubeconfig_<cluster_name>`.

Reviewer 2: ```suggestion
You can interact with the EKS cluster using `kubectl` and `helm` with the kubeconfig file `credentials/kubeconfig_<cluster_name>`:
```

helm ls
```

# Destroy

It may take some while to finish destroying the cluster.

> ``` shell

Reviewer: @lilin90 Is this space necessary?

helm ls
```

> ## Monitoring

Reviewer 2: ```suggestion
## Monitor
```
Unify the wording

You can access the `monitor_endpoint` address (printed in outputs) using your web browser to view monitoring metrics.

The initial Grafana login credentials are:

>     - User: admin

Reviewer 2: ```suggestion

- User: admin
```

>     - Password: admin

Reviewer 2: ```suggestion
- Password: admin
```

> # Destroy

Reviewer 2: ```suggestion
## Destroy
```

It may take some while to finish destroying the cluster.

``` shell
$ terraform destroy
```

> **Note:** You have to manually delete the EBS volumes in AWS console after running `terraform destroy` if you do not need the data on the volumes anymore.

> ## Upgrade TiDB cluster

Reviewer 2: ```suggestion
## Upgrade
```

To upgrade the TiDB cluster, modify the `tidb_version` variable to a higher version in the `variables.tf` file, and then run `terraform apply`.

> *Note*: The upgrading doesn't finish immediately. You can watch the upgrading process by `kubectl --kubeconfig credentials/kubeconfig_<cluster_name> get po -n tidb --watch`.

> ## Scale TiDB cluster

Reviewer 2: ```suggestion
## Scale
```

> To upgrade TiDB cluster, modify `tidb_version` variable to a higher version in variables.tf and run `terraform apply`.

Reviewer 2: ```suggestion
To upgrade the TiDB cluster, modify the `tidb_version` variable to a higher version in the `variables.tf` file, and then run `terraform apply`.
```

> *Note*: The upgrading doesn't finish immediately. You can watch the upgrading process by `kubectl --kubeconfig credentials/kubeconfig_<cluster_name> get po -n tidb --watch`.

## Scale TiDB cluster

> To scale TiDB cluster, modify `tikv_count` or `tidb_count` to your desired count, and then run `terraform apply`.

Reviewer 2: ```suggestion
To scale the TiDB cluster, modify the `tikv_count` or `tidb_count` variable to your desired count in the `variables.tf` file, and then run `terraform apply`.
```

> *Note*: Currently, scaling in is not supported since we cannot determine which node to scale. Scaling out needs a few minutes to complete, you can watch the scaling out by `kubectl --kubeconfig credentials/kubeconfig_<cluster_name> get po -n tidb --watch`.

## Customize

> You can change default values in `variables.tf` (like the cluster name and image versions) as needed.

Reviewer 2: ```suggestion
You can change default values in `variables.tf` (such as the cluster name and image versions) as needed.
```
